### ISSUE TITLE
Responsive Player

### ISSUE THREAD

**Issue by dhbdev** [justinhunt/moodle-filter_videoeasy#2]
Hi Justin,

first, thank you for your plugin. It is very helpful.
I have one question:
Is it possible to configure the player window also responsive?
If I login to a Moodle course with videos in an text area, the player window isn't responsive. It is horizontal scrollable.

Thanks for an answer
Andreas

**Comment by justinhunt** [comment]
Unlike other video filters for Moodle, you are completely free to customize the way the videos are embedded on the page. Possibly some of the pre-made templates are not responsive, but you can modify those settings, or choose a different player template.

**Comment by dhbdev** [comment]
Hi Justin,

thank you for your fast reply.
I play a little while with some options, but it still don’t work.

This is the template code I use:
“<video id="@@AUTOID@@" class="video-js vjs-default-skin" controls preload="auto" width="@@WIDTH@@" height="@@HEIGHT@@" poster="@@AUTOPOSTERURLPNG@@"  data-setup='{"example_option":true}'><source src="@@VIDEOURL@@" type="@@AUTOMIME@@" /></video>”
For every video I use a individual poster.

This ist the HTML Code, which is generated by the template:
“<div id="fv_144915588268953973" class="video-js vjs-default-skin vjs-paused vjs-controls-enabled vjs-user-inactive" preload="auto" poster="http://studistartup.dhbw-stuttgart.de/moodle/pluginfile.php/21/mod_label/intro/division_von_bruechen.png" data-setup="{"example_option":true}" style="width: 480px; height: 280px;">
<video id="fv_144915588268953973_html5_api" class="vjs-tech" data-setup="{"example_option":true}" poster="http://studistartup.dhbw-stuttgart.de/moodle/pluginfile.php/21/mod_label/intro/division_von_bruechen.png" preload="auto">

<div></div>

<div class="vjs-poster" tabindex="-1" style="background-image: url("http://studistartup.dhbw-stuttgart.de/moodle/pluginfile.php/21/mod_label/intro/division_von_bruechen.png");"></div>

<div class="vjs-text-track-display"></div>

<div class="vjs-loading-spinner"></div>

<div class="vjs-big-play-button" role="button" aria-live="polite" tabindex="0" aria-label="play video">
<div class="vjs-control-bar">
<div class="vjs-error-display">
</div>"

The <div> is generated automatically by your filter plugin I assume.
The section style="width: 480px; height: 280px; in the div tag comes from the section “Video JS defaults” If I changed “width” to "max-width:480px” with firebug in firefox the player is responsive in the width of the video. But if I set max-height the player window collapses.
How can I customize the div settings to have both width and height responsive?
I could not set any of the options in the template of the VideJS player.

Thanks again for an answer and a tipp
Andreas

**Comment by justinhunt** [comment]
The width=640 and height=480 come from the defaults that are set for the template in the field "Video JS defaults" (this is on the video js template's settings page ... scroll down to see it)

To make video js responsive , you should follow this guide http://coolestguidesontheplanet.com/videodrome/videojs
and try and implement it in videoeasy. 

Other player templates might be easier than this. You could try flowplayer for example.

**Comment by dhbdev** [comment]
Hi,

O.K have it.
I’m able to set WIDTH=80% i.e and with Custom CSS I’m able with the class “.video-js” to set some custom css options:
.video-js  {
/_padding-top:1%;_/
max-width:480px;
background-color: #fff;
}
Now the players width is responsive, but if I set the height to “auto” the player window inside Moodle collapsed.
So for now I’m only able to set a fix height, but this is not optimal.
Do you have a last tipp, how to fix the height issue, if I set it to auto?
Additionally it isn’t possible to set the height to a % level.

Thanks again
Andreas

**Comment by justinhunt** [comment]
Thanks for sticking with it. I can't wait to see the finished template.
What happens if you don't specify height at all?

**Comment by dhbdev** [comment]
Hi Justin,

if I don’t specify height at all it sets height to 150px:
<div id="fv_14492144801716726630" class="video-js vjs-default-skin vjs-paused vjs-controls-enabled vjs-user-inactive" preload="auto" poster="http://studistartup.dhbw-stuttgart.de/moodle/pluginfile.php/37/mod_label/intro/addition_von_bruechen_hb.png" data-setup="{"example_option":true}" style="width: 100%; height: 150px;">

**Comment by justinhunt** [comment]
Try this. create a new template based on the video.js preset:
 Replace the following sections:

# Template

<div class="video-js-container" style="width: 80%;max-width: 640px">
<video id="@@AUTOID@@" class="video-js vjs-default-skin" controls
preload="auto" width="@@WIDTH@@" height="@@HEIGHT@@"
data-setup='{"example_option":true}'><source src="@@VIDEOURL@@"
type="@@AUTOMIME@@" /></video>
</div>

# 

# Defaults

# WIDTH=auto,HEIGHT=auto

# Custom CSS

.video-js {padding-top: 56.25%}

# .vjs-fullscreen {padding-top: 0px}

How does that work?

**Comment by dhbdev** [comment]
hi Justin,

many many thanks :-)
It works like a charme.

I think it is a great plugin.
If you make it possible to enhance the Atto/TinyMCE Editor with options which make it easier to integrate poster images and perhaps some options for sizes and so on for “normal” users in Moodle than it is a really great plugin in the world of third party plugins for Moodle.
The last wish is an option for generating playlists in an easy and fast way :-) without a deep knowledge of the videojs infrastructure.

Thanks again and a nice weekend
Andreas

**Comment by justinhunt** [comment]
Indeed. Some of that stuff can be done easier with the Generico filter/atto plugin right now. But you can pass in parameters to videoeasy via the url, and they will be accessible in the your template.
e.g < a href="http://mysite.com/myvideo.mp4?fruit=banana&animal=cat" >video< /a >
Would throw up @@fruit@@ and @@animal@@ as variables. Thats one way to handle it. Not very GUI though.

Also Generico has a preset template for a video playlist/gallery. See here:
http://demo.poodll.com/mod/page/view.php?id=146

**Comment by dhbdev** [comment]
O.K looks good.
I test the Generico filter  and Atto enhancement next week and give you feedback.

**Comment by dhbdev** [comment]
Hi Justin,

after my christmas holidays I’m back on stage ;-)
Hope you’re well and I wish you all the best and a happy new year.

As you suggested I installed the Generico plugin on my Moodle site.
Actually I use Moodle 2.7.10+ as for compatibility reasons.

After installing the Generico plugin I couldn’t configure the template settings for the video gallery and got the following error code:
Fatal error: Call to undefined method page_requirements_manager::js_call_amd() in /var/www/html/moodle/filter/generico/locallib.php on line 250

After commenting out line 250 I was able to configure the Template but it doesn’t work.
The videos are shown like in the attached screenshot without the playlist.

Four questions.
1. Do I need the AMD loading mechanism? I saw it is only usable on Moodle 2.9.x onwards
2. Do I need the JW player instead of VideoJS?
3. Is the above error a bug oder do I need an older release of the Generico plugin?
4. Is it possible to have posters for the videos also with playlists and the Genericoplugin?

Thanks
Andreas

**Comment by dhbdev** [comment]
…sorry, now with the screenshot….

Hi Justin,

after my christmas holidays I’m back on stage ;-)
Hope you’re well and I wish you all the best and a happy new year.

As you suggested I installed the Generico plugin on my Moodle site.
Actually I use Moodle 2.7.10+ as for compatibility reasons.

After installing the Generico plugin I couldn’t configure the template settings for the video gallery and got the following error code:
Fatal error: Call to undefined method page_requirements_manager::js_call_amd() in /var/www/html/moodle/filter/generico/locallib.php on line 250

After commenting out line 250 I was able to configure the Template but it doesn’t work.
The videos are shown like in the attached screenshot without the playlist.

Four questions.
1. Do I need the AMD loading mechanism? I saw it is only usable on Moodle 2.9.x onwards
2. Do I need the JW player instead of VideoJS?
3. Is the above error a bug oder do I need an older release of the Generico plugin?
4. Is it possible to have posters for the videos also with playlists and the Genericoplugin?

Thanks
Andreas

**Comment by justinhunt** [comment]
The AMD stuff only works on Moodle 2.9 and above. I have to apologize
because I have not kept Generico backwards compatible with the Moodle 2.8
and earlier. I should really maintain a pre 2.9 branch.

I suggest that for now, you use the version that is available from here:
https://moodle.org/plugins/filter_generico until you can move on Moodle 2.9
or 3.x

Justin
